src/app/collection/config/tab/attribute-list/attribute-list.component.ts

```diff
@@ -191,7 +191,7 @@ export class AttributeListComponent {
 
     this.suggestions = Const.constraints.map(flattenConstraints)
       .reduce(flatten, [])
-      .filter(suggestion => startsWith(suggestion.name, this.activeAttribute.newConstraint))
+      .filter(suggestion => startsWith(suggestion.name, this.activeAttribute['newConstraint']))
```

Comment:
Why do we need to keep this `newConstraint` field on an attribute? In general, we should not modify existing objects since they can come from the store and we would like to be able to access older versions of data in order to make application behavior predictable and debugging much easier.